content-src/components/DiscoveryStreamBase/DiscoveryStreamBase.jsx

```diff
@@ -170,14 +170,11 @@ export class _DiscoveryStreamBase extends React.PureComponent {
   render() {
     // Select layout render data by adding spocs and position to recommendations
     const {layoutRender, spocsFill} = selectLayoutRender(this.props.DiscoveryStream, this.props.Prefs.values, rickRollCache);
-    const {config, feeds, spocs} = this.props.DiscoveryStream;
-    if (!spocs.loaded || !feeds.loaded) {
-      return null;
-    }
+    const {config, spocs, feeds} = this.props.DiscoveryStream;
 
     // Send SPOCS Fill if any. Note that it should not send it again if the same
     // page gets re-rendered by state changes.
-    if (spocsFill.length && !this._spocsFillSent) {
+    if (spocs.loaded && feeds.loaded && spocsFill.length && !this._spocsFillSent) {
```

Comment:
When the subsequent feed loads complete and trigger the state changes, which in turn might generate some new spocs fill pings. Is it possible that those pings would never get sent if it has already sent one earlier?

Reply:
Hm, not sure I follow, but the intended solution here is to not send a ping until everything is loaded, it would then go ahead and fire the ping, and set `_spocsFillSent` to true, then next time it comes in here, it would again see it as loaded, but this time `_spocsFillSent` is true, so it won't send again.

Reply:
Ah, I see. That's why you've added those `loaded` checks here. I was wondering if we can do the same check inside of `selectLayoutRender` [here](https://github.com/mozilla/activity-stream/blob/master/content-src/lib/selectLayoutRender.js#L115). I.e. if either SPOCS or feeds has not been fully loaded, then don't bother calculating the SPOCS Fill at all.

Reply:
I'll give that a try. Should work. :+1: